static/js/editor.ts

```diff
@@ -0,0 +1,106 @@
+export type Breakpoints = Record<number, string>;
+
+
+export interface HedyEditorCreator {
+  // TODO: Not sure yet if it should return a HedyEditor or change local variable names
+  /**
+   * This function should initialize the editor and set up all the required
+   * event handlers
+   * @param $editor reference to the div that contains the main editor
+   */
+  initializeMainEditor: ($editor: JQuery) => HedyEditor | undefined;
+  /**
+   * 
+   * @param element the element that will contain this editor
+   * @param isReadOnly to decide weather to remove the cursor
+   * @param isMainEditor should we show the line numbers
+   */
+  turnIntoEditor: (element: HTMLElement, isReadOnly: boolean, isMainEditor: boolean) => HedyEditor;
+
+  /**
+   * Ininitialize an editor that appears in a modal
+   * @param $editor reference to the div that contains this editor
+   */
+  initializeModalEditor: ($editor: JQuery) => HedyEditor;
+}
+
+export interface HedyEditor {
+  /**
+ * Represents whether there's an open 'ask' prompt
+ */
+  askPromptOpen: boolean;
+  /**
+   * Set the highlither rules for a particular level
+   * @param level      
+   */
+  setHighliterForLevel: (level: number) => void;
```

Comment:
> "highlight" upside_down_face

Nice catch! 

> This declares a function-valued member, which of course works. But, you will also be able to _assign_ the member if you declare it like this, because it's a read/write field.
> 
> If you want to declare a method, it looks like this:
> 
> ```ts
> interface HedyEditor {
>   setHighlighterForLevel(level: number): void;
> }
> ```

Thanks for the tip!

> Isn't this function more about setting a level than setting a highlighter?

You mean it should be called, `setLevel`? I think so, but maybe `setHighlighterForLevel` is more explicit?